python/cuspatial/cuspatial/core/interpolate.py

```diff
@@ -67,9 +67,9 @@ class CubicSpline:
     Note
     ----
     cuSpatial will outperform scipy when many splines are
-    fit simultaneously. Data must be arranged in a SoA format, and the
-    exclusive `prefix_sum` of the separate curves must also be passed to the
-    function. See example for detail.
+    fit simultaneously. Data must be arranged in a structure of arrays (SoA)
+    format, and the exclusive `prefix_sum` of the separate curves must also be
```

Comment:
These are the offsets to the . Don't call it a prefix sum, call it offsets. Prefix sum would be one possible way of computing these offsets, and is an implementation detail. The example below does not compute a prefix sum, it just names a variable `prefix_sum`. The example uses fixed-length splines so the offsets are all just multiples of that length.

Apologies, I know this example has been around for a while, but I think it is not very self-explanatory. It is made up of a bunch of fairly opaque numpy calls to generate data. If instead the example used trivial length splines (e.g. 4 or 5 points), then the data could be explicitly written out and the example would be easier to follow.

Reply:
Fair enough, I updated the example to make it clearer.